projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -478,29 +491,62 @@ export function addConstructorParam(
   if (!constructorNode) {
     throw new SchematicsException(`No constructor found in ${sourcePath}.`);
   }
-
   const changes: Change[] = [];
-
   if (!isInjected(constructorNode, paramToAdd)) {
     changes.push(
       injectService(
         constructorNode,
         sourcePath,
         paramToAdd.className,
-        'no-modifier'
+        'no-modifier',
+        undefined,
+        paramToAdd.staticType,
+        paramToAdd.injectionToken?.token,
+        paramToAdd.injectionToken?.isArray
       )
     );
   }
 
-  if (!isImported(source, paramToAdd.className, paramToAdd.importPath)) {
-    changes.push(
-      insertImport(
-        source,
-        sourcePath,
-        paramToAdd.className,
-        paramToAdd.importPath
-      )
-    );
+  if (paramToAdd.importPath) {
+    if (!isImported(source, paramToAdd.className, paramToAdd.importPath)) {
+      changes.push(
+        insertImport(
+          source,
+          sourcePath,
+          paramToAdd.className,
+          paramToAdd.importPath
+        )
+      );
+    }
+  }
+  if (paramToAdd.injectionToken?.token) {
+    if (!isImported(source, INJECT_DECORATOR, ANGULAR_CORE)) {
+      changes.push(
+        insertImport(source, sourcePath, INJECT_DECORATOR, ANGULAR_CORE)
+      );
+    }
+
+    if (
+      paramToAdd.injectionToken.importPath &&
+      paramToAdd.injectionToken.token !== paramToAdd.className
```

Comment:
Why this check?

Reply:
Same as the other check bellow, I added a comment.